### DISCUSSION TITLE
I think I have some great ideas for your Identity mod!!! I put a lot of time, thought, and effort into this! It would mean so much if you read this and concidered it!!! I'm sure it would greatly improve the mod!!!  :D

### DISCUSSION THREAD

**Discussion by CreeprVictor** [Draylar/identity#638] [General]
### I think I have some great ideas for your Identity mod!
**please read this, I put a lot of time, thought, and effort into this. If you see this, it would be greatly appreciated for you to read this and atleast reply to this**
Hey, I've been playing with Your Identity mod, good mod! 

I have some ideas to add more to it!

- Selecting Mobs & Missing mob variants: I've noticed I'm not able to select some mob variants, such as the other rabbit types, mabey, to reduce clutter and solve this issue, you could add like a drop down thing for some mobs to select different variants of the mob! (Like the creative inventory in BE). And for more complex mob variations, like horses and tropical fish (Or have this be the default as opposed to the dropdown), you'd have a GUI to where you can chose/edit the variant! Where you would have to kill the mob with the variant to get the variant! So, if you killed a black horse with spots, you can only become that, kill a white one with no decal, you could now be like a black horse with no decal, and vise versa!

- Mob Abilities: Some mobs are missing some abilities, such as guardian lasers, evoker spawning vexes, vexes going through walls, etc.  Now, I know, there is only 1 ability button, but, that is where # 3 comes in!

- Food: Some morphs should be able to eat their foods, like cows eat wheat, but not meat, etc.

- Adding More Ability Keys, using 1-9 keys: Instead of being restricted to just one, we use the number keys 1-9! While morphed into a mob, you instead use the number keys for abilities! (Talked about more under "Possibilities" and under "Abilities"). And some abilities would require you to hold the ability button (Like creeper and blaze fireballs)! And they could be shown in a 2nd hotbar above the original!

- Vision Shaders: When morphed into a mob like a creeper, enderman, or spider, you get the shader applied like when you spectate them!  (mabey even add some to other mobs to show how their real-world counterparts see!)

### Possibilities
With all the stuff here, you would really more so be able to really get the experience of morphing into another mob!

Now we can add more abilities, like:

- Evoker: 1= Evocation fangs 2= Vexes 3(mabey)= Convert (Blue sheep to red)

- Blaze: 1= Shoot fireballs 2= toggle float down/hover/float up

- Etc, more is shown under "Abilities"

### Abilities

- Axolotl: Regeneration
- Bee: Pollinate
- Blaze: Shoot 5 fireballs instead of 1
- Blaze: toggle float down/hover/float up
- Camel: Dash (by holding jump)
- Chicken falls slowly
- Chicken: Lay egg
- Creeper: Explode(which you have)
- Creeper has a creeper's vision
- Ender Dragon: Shoot dragon breath ball (which you have)
- Ender Dragon: Breath dragon breath
- Enderman: Teleport (which you have)
- Enderman: Random teleport (like chorus fruit)
- Enderman: Ender pearl
- Enderman has inverted vision
- Evoker: Fangs (which you have)
- Evoker: Vexes
- Evoker(mabey): Convert(Blue sheep to red)
- Fox: Dive
- Ghast: Shoot fireball (which you have)
- Goat: Ram
- Goat(Mabey): Precision Jump (hold it, and it will show you where you can jump to, release, and you jump there! This would work similar to in the goat's AI)
- Guardian (&Elder): Guardian laser
- Guardian (&Elder) have thorns effect (which you have)
- Hoglin zombifies
- Horse, donkey, & mule: hold jump for jump like when riding a horse
- Illusioner: Illusion
- Panda (Playful): Roll
- Piglin zombifies
- Pufferfish: Puff up
- Ravager: Roar
- Ravager breaks leaves and crops
- Sheep: Eat grass, grow wool
- Sheep is shearable
- Shulker is stuck in a fixed position aligned with blocks
- Shulker can be on the sides and bottom of blocks
- Shulker: Teleport
- Shulker: Teleport Random
- Shulker: Open/Close Shell
- Shulker: Shoot shulker bullet (only when shell is open)
- Silverfish: Hide in stone
- Slime and magma cube split on death (mabey)
- Sniffer: Dig
- Snow golem: Shoot (only 1) snowball
- Snow golem leaves snow trail
- Spider climbs walls (which you have)
- Spider doesn't get slowed by cobwebs
- Spider has spider vision
- Squid (&Glow squid): Ink jet
- Strider walks on lava (which you have)
- Strider gets cold out of lava
- Tropical fish can be edited to be any of the [3,072 different unique combinations by choosing any of 2 shapes, 6 patterns, 16 base colors, and 16 pattern colors. Also, having presets of the 22 preset designs](https://minecraft.wiki/w/Tropical_Fish#Varieties)
- Vex: Aggro (shows the red attacking skin)
- Vex can go through blocks (and is forced into fly mode)
- Warden: Sonic Boom (which you have)
- Warden (mabey): Sniff (shows nearby mobs, including players)
- Warden (mabey) sees vibrations
- Warden (mabey) uses the "Sobel" shader that was featured in the "Super Secret Settings" a while back. ([link to a mod that adds this in newer versions](https://www.curseforge.com/minecraft/mc-mods/super-secret-settings))
- Witch: Drink potion
- Witch: Throw negative potion
- Witch: Throw helpful potion
- Wither: Shoot wither skull (which you have)
- Wither: Shoot blue wither skull
- Wither: Toggle shield
- Wither is unaccessable in survival mode without commands (can be bypassed by going into creative mode, morphing, then survival)
- Wolf: Aggro (shows aggressive skin)

### Other and bugs
Other things I've noticed include:

- Silverfish morph can't hide in stone
- Shulker morph not restricted to block alignment
- Shulker morph can walk
- Piglin don't know if to attack you or not when you are a piglin and aren't wearing gold, they freak out.
- Piglin also freak out whenever you are morphed not wearing gold for the same reason
- Piglin and hoglin morphs don't convert to zombies
- Fox morph can't dive
- Goat morph can't charge
- Blaze morph only shoots 1 fireball
- Horse, donkey, and mule morph has normal player speed, jump, and can't auto step up 1 block
- Horse, panda and rabbit are missing their variants
- No brown mooshroom
- Bee morph can't pollinate
- Dolphin morph dies shortly on land
- Dolphin morph can't ever drown
- Axolotl morph never dries out
- Guardian and elder guardian morphs die on land
- Phantoms aren't scared of cat or ocelot morph
- Creepers aren't scared of cat morph and aren't scared of ocelot morph in creative mode
- Foxes are scared of morphed players
- among other morphs not being scared of morphed they should be
- Sniffer, axolotl on land, and other morphs have player speed
- Chicken doesn't have slow falling
- Ravager morph doesn't break leaves and crops
- Spider morphs get stuck in cobwebs
- Creeper morph explosion doesn't damage blocks and you can't defuse yourself (I also think you should take some damage, or atleast be affected by the explosion, dealing you fall damage)
- Endermite and silverfish sink all the way through 1 soul sand block
- Foxes don't attack chicken morph
- Polar bear doesn't attack fox morph
- I could be wrong, but I don't think witches heal illager morphs in raids
- Wither skeleton doesn't attack piglin morph
- Piglin groups don't attack wither skeleton morph
- Strider morph warm even when not on lava
- Ravager can't roar
- Vex takes fall damage

**Comment by Natedogwoooof** [comment]
Use woodwalkers and recorded its higher quality

**Comment by Natedogwoooof** [comment]
Remorphed

**Reply by CreeprVictor** [reply]
Tried it, it's not really any better, it only adds a couple more things and is more inconvenient, like you need to use a separate key to morph back
